### ISSUE TITLE
Broken many0! example in README

### ISSUE THREAD

**Issue by rgardner** [Geal/nom#128]
I was trying to test out the many0! example you have in the README:
```rust
#[macro_use]
extern crate nom;
use nom::IResult::{Done};
fn main() {
    named!(multi, many0!( tag!( "abcd" ) ) );
    let a = b"abcdef";
    let b = b"abcdabcdef";
    let c = b"azerty";
    assert_eq!(multi(a), Done(b"ef", vec![b"abcd"]));
    assert_eq!(multi(b), Done(b"ef", vec![b"abcd", b"abcd"]));
    assert_eq!(multi(c), Done(b"azerty", Vec::new()));
}
```

However, I get this:

```
<std macros>:5:22: 5:33 error: mismatched types:
 expected `nom::internal::IResult<&[u8], &[u8]>`,
    found `nom::internal::IResult<&[u8; 2], collections::vec::Vec<&[u8; 4]>, _>`
(expected slice,
    found array of 2 elements) [E0308]
<std macros>:5 if ! ( * left_val == * right_val ) {
                                    ^~~~~~~~~~~
src/main.rs:9:5: 9:54 note: in this expansion of assert_eq! (defined in <std macros>)
<std macros>:5:22: 5:33 help: run `rustc --explain E0308` to see a detailed explanation
<std macros>:5:22: 5:33 error: mismatched types:
 expected `nom::internal::IResult<&[u8], &[u8]>`,
    found `nom::internal::IResult<&[u8; 2], collections::vec::Vec<&[u8; 4]>, _>`
(expected slice,
    found array of 2 elements) [E0308]
<std macros>:5 if ! ( * left_val == * right_val ) {
                                    ^~~~~~~~~~~
src/main.rs:10:5: 10:63 note: in this expansion of assert_eq! (defined in <std macros>)
<std macros>:5:22: 5:33 help: run `rustc --explain E0308` to see a detailed explanation
<std macros>:5:22: 5:33 error: mismatched types:
 expected `nom::internal::IResult<&[u8], &[u8]>`,
    found `nom::internal::IResult<&[u8; 6], collections::vec::Vec<_>, _>`
(expected slice,
    found array of 6 elements) [E0308]
<std macros>:5 if ! ( * left_val == * right_val ) {
                                    ^~~~~~~~~~~
src/main.rs:11:5: 11:55 note: in this expansion of assert_eq! (defined in <std macros>)
<std macros>:5:22: 5:33 help: run `rustc --explain E0308` to see a detailed explanation
<nom macros>:6:38: 6:41 error: mismatched types:
 expected `&[u8]`,
    found `collections::vec::Vec<&[u8]>`
(expected &-ptr,
    found struct `collections::vec::Vec`) [E0308]
<nom macros>:6 } $ crate:: IResult:: Done ( input , res ) } ) ; ( $ i : expr , $ f : expr )
                                                    ^~~
<nom macros>:20:1: 20:34 note: in this expansion of many0! (defined in <nom macros>)
src/main.rs:5:5: 5:46 note: in this expansion of named! (defined in <nom macros>)
<nom macros>:6:38: 6:41 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to 4 previous errors
Could not compile `cookie`.

To learn more, run the command again with --verbose.
```

rustc 1.6.0-nightly (9303055f3 2015-11-19)
nom = "1.0.0"

I was able to get the intended functionality (I think?) by changing the example to:
```rust
#[macro_use]
extern crate nom;
use nom::IResult::{Done};
fn main() {
    named!(multi<&[u8], Vec<&[u8]> >, many0!( tag!( "abcd" ) ) );
    let a = b"abcdef";
    let b = b"abcdabcdef";
    let c = b"azerty";
    assert_eq!(multi(a), Done(&b"ef"[..], vec![&b"abcd"[..]]));
    assert_eq!(multi(b), Done(&b"ef"[..], vec![&b"abcd"[..], &b"abcd"[..]]));
    assert_eq!(multi(c), Done(&b"azerty"[..], Vec::new()));
}
```

It's syntactically heavier than your example, so hopefully there's a less verbose way of accomplishing this.

**Comment by Geal** [comment]
Unfortunately, that's the common representation for byte slices from static strings. I can add a conversion to`&str`,  though, that would be nicer.

**Comment by Geal** [comment]
Fixed in fae037a. Is the example alright for you?

**Comment by rgardner** [comment]
LGTM, thanks for fixing it. Also, thanks for a great project!
